src/core/experimental/memory/memcpy.d

```diff
@@ -0,0 +1,401 @@
+/**
+ * Pure D replacement of the C Standard Library memcpy().
+ * There is an idiomatic-D interface, memcpy(), which is split into 3 overloads.
+ * One taking static types, one dynamic arrays and one static arrays.
+ * Also, there is available a C-like interface, the Dmemcpy() (which is named Dmemcpy
+ * for disambiguation with the C memcpy() a _similar_ interface) that
+ * is the classic (void*, void*, size_t) interface.
+ * N.B.: Both the memcpy() here and Dmemcpy() return nothing, contrary to the C Standard
+ * Library version.
+ * Source: $(DRUNTIMESRC core/experimental/memory/memcpy.d)
+ */
+module core.experimental.memory.memcpy;
+
+import core.internal.traits : isArray;
+
+/**
+ * Handle Static Types
```

Comment:
This distinction isn't important to the user.

Reply:
You mean it should be just a `//` that is not part of the doc?

Reply:
No I meant that as you have both overloads defined, this is an implementation detail. Though as said on another point there's a case to simplify the interface even further, but it definitely shouldn't expose such details to the user.

Reply:
> this is an implementation detail
Yes, so the `//` wouldn't be the right choice i.e. not be part of the doc?
On the merging of the interfaces there were problems that I could not solve: https://github.com/dlang/druntime/pull/2687#discussion_r311488051